src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonPropertyInfo.cs

```diff
@@ -264,6 +282,12 @@ internal static JsonPropertyInfo GetPropertyPlaceholder()
         /// </summary>
         public Type PropertyType { get; }
 
+        internal void MarkAsConstructorParameter()
```

Comment:
Do we really need this method? `IsConstructorParameter` has a private setter so we can assume that the assert invariant is preserved.

Reply:
This is called from JsonTypeInfo.CreateConstructorParameter - I prefer to have it here rather than making `IsConstructorParameter` internal - this way I can add extra check and make sure it's not going to be misused in the future.